nwb_conversion_tools/utils/nwbephyswriterdatachunkiterator.py

```diff
@@ -1,48 +1,98 @@
+          The upper bound on size in gigabytes (GB) of each selection from the iteration.
+          The buffer_shape will be set implicitly by this argument.
+          The default is 1GB.
+        buffer_shape : tuple, optional
+          If the user desires further control over the shaping of the buffer over all the axes,
+          they may set this explicitly as a tuple of the same length as the data shape indicating the desired shape
+          of each selection.
+          The default is None.
+        chunk_mb : float, optional
+          The upper bound on size in megabytes (MB) of each internally set chunk for the HDF5 dataset.
+          The chunk_shape will be set implicitly by this argument.
+          The default is 1MB, which is highly recommended by the HDF5 group. For more details, see
+          https://support.hdfgroup.org/HDF5/doc/TechNotes/TechNote-HDF5-ImprovingIOPerformanceCompressedDatasets.pdf
+        chunk_shape : tuple, optional
+          If the user desires further control over the shaping of the buffer over all the axes,
+          they may set this explicitly as a tuple of the same length as the data shape indicating the desired shape
+          of each selection.
+          The default is None.
+        """
         self.segment_index = segment_index
         self.write_scaled = write_scaled
         self.ephys_writer = ephys_writer
         self.channel_ids = list(ephys_writer.recording.get_channel_ids())
         self.unsigned_coercion = (
-            np.zeros([len(self.channel_ids)]) if unsigned_coercion is None else np.array(unsigned_coercion)
+            np.zeros(shape=(len(self.channel_ids)), dtype=int)
+            if unsigned_coercion is None
+            else np.array(unsigned_coercion).astype(int)
         )
+        self._dtype = np.dtype(dtype)
         super().__init__(buffer_gb=buffer_gb, buffer_shape=buffer_shape, chunk_mb=chunk_mb, chunk_shape=chunk_shape)
 
     def _get_data(self, selection: Tuple[slice]) -> Iterable:
         channel_ids = self.channel_ids[selection[1]]
         channel_idxs = np.array([self.channel_ids.index(ch) for ch in channel_ids], dtype="int")
-        return (
-            self.ephys_writer._get_traces(
-                channel_ids=channel_ids,
-                start_frame=selection[0].start,
-                end_frame=selection[0].stop,
-                return_scaled=self.write_scaled,
-                segment_index=self.segment_index,
-            )
-            + self.unsigned_coercion[channel_idxs]
+
+        data = self.ephys_writer._get_traces(
+            channel_ids=channel_ids,
+            start_frame=selection[0].start,
+            end_frame=selection[0].stop,
+            return_scaled=self.write_scaled,
+            segment_index=self.segment_index,
         )
 
+        if self.dtype is not None:
+            data = data.astype(self.dtype)
+
+        data = data + self.unsigned_coercion[channel_idxs]
+        return data
```

Comment:
Note that it's possible, the way it's currently setup, for the modification by the coercion to promote the `dtype` in a way that wouldn't return the explicitly set `dtype`. This links to the redundancy problem mentioned above when using this Iterator in a really arbitrary way.

I.e., `NwbEphysWriterDataChunkIterator(ephys_writer=my_writer_with_unit8_data, dtype="int8", unsigned_coercion=[-(2**15+1)]*len(series), return_scaled=True)` is a problem because what gets returned at each buffer is none of those things requested by the arguments.